server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -4336,6 +4365,174 @@ public SSHKeyPair registerSSHKeyPair(final RegisterSSHKeyPairCmd cmd) {
         return createAndSaveSSHKeyPair(name, fingerprint, publicKey, null, owner);
     }
 
+    @Override
+    public boolean deleteUserData(final DeleteUserDataCmd cmd) {
```

Comment:
would it be better to have `UserDataManager` to manage userdata actions ?

Reply:
Thats a good suggestion but I think going forward it is better to keep one separate manager for all userdata services in network offering like passwords, userdata, metada. So I'm keeping this here.